system/jlib/jptree.cpp

```diff
@@ -7614,13 +7672,58 @@ static constexpr const char * currentVersion = "1.0";
  *     '__remove__' then that element is removed, otherwise it is merged.  If there is no match it is added.
 */
 
+static bool checkInSequence(IPropertyTree & child, StringAttr &seqname, bool &first, bool &endprior)
+{
+    first = false;
+    endprior = false;
+    if (seqname.length() && streq(seqname, child.queryName()))
+        return true;
+    endprior = !seqname.isEmpty();
+    if (child.isArray(nullptr))
+    {
+        first=true;
+        seqname.set(child.queryName());
+        return true;
+    }
+    seqname.clear();
+    return false;
+}
+
+inline bool isScalarItem(IPropertyTree &child)
+{
+    if (child.hasChildren())
+        return false;
+    return static_cast<PTree*>(&child)->getAttributeCount()==0;
```

Comment:
@ghalliday To avoid this I could add either getAttributeCount() or isScalar() to IPropertyTree interface.  If either would be preferred to this being PTree internal.

Reply:
or I can instantiate an attribute iterator but that's a lot of overhead each time.